options/options_test.go

```diff
@@ -198,6 +198,21 @@ func TestProcessAppConfig(t *testing.T) {
 		require.Error(t, options.ProcessAppConfig(testService, "core-data"))
 	})
 
+	t.Run("reject unknown app", func(t *testing.T) {
+		const key, value = "apps.unknown.config.x-y", "value"
+
+		require.NoError(t, snapctl.Set(appOptions, "true").Run())
+		require.NoError(t, snapctl.Set(key, value).Run())
```

Comment:
why isn't it `require.Error(t, snapctl.Set(key, value).Run())`?

Reply:
The snap that is used for these tests doesn't have a configure hook. Using `snapctl` (instead of `snap`) will not trigger the configure hook even if there was one. It simply sets the values without doing anything else. There will be error only if the key is invalid accroding to snapd.

EdgeX snaps have a configure hook which is triggered after the values are set (when using `snap`). The configure hooks call the `options.ProcessAppConfig(...)` function which actually does the processing. In these tests, we call that function manually (see line 211). That is when the error is raised.

Reply:
Thank you for the explanation!